test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go

```diff
@@ -550,27 +540,55 @@ func setup(ctx context.Context) func(ctx2 context.Context) {
 			By(fmt.Sprintf("skipping teardown - no changes to profile %q were applied", profile.Name))
 		}
 	}
-	testprofiles.UpdateWithRetry(profile)
-	mcp, err := mcps.GetByProfile(profile)
-	Expect(err).ToNot(HaveOccurred())
-
-	mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
-	mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+	profiles.UpdateWithRetry(profile)
+	poolName := getPoolName()
+	CheckProfileUpdating(poolName, profile)
 
 	teardown := func(ctx2 context.Context) {
 		By(fmt.Sprintf("executing teardown - revert profile %q back to its initial state", profile.Name))
-		Expect(testclient.Client.Get(ctx2, client.ObjectKeyFromObject(initialProfile), profile))
-		testprofiles.UpdateWithRetry(initialProfile)
+		Expect(testclient.ControlPlaneClient.Get(ctx2, client.ObjectKeyFromObject(initialProfile), profile))
+		profiles.UpdateWithRetry(initialProfile)
 
 		// do not wait if nothing has changed
 		if initialProfile.ResourceVersion != profile.ResourceVersion {
-			mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
-			mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+			mcps.WaitForCondition(poolName, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+			mcps.WaitForCondition(poolName, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
 		}
 	}
 	return teardown
 }
 
+func CheckProfileUpdating(poolName string, profile *performancev2.PerformanceProfile) {
+	if hypershift.IsHypershiftCluster() {
```

Comment:
`profilesupdate.WaitForTuningUpdating` already doing that and can be used for both HyperShift and OpenShift

Reply:
Resolved by replacing with `profilesupdate.WaitForTuningUpdating` at each place.